src/utils.jl

```diff
@@ -178,7 +178,7 @@ function validate_inputs(x, y)
     if dim(x) != dim(y) # Passes by default if `dim` is not defined
         throw(
             DimensionMismatch(
-                "Dimensionality of x ($(dim(x))) not equality to that of y ($(dim(y)))"
+                "Dimensionality of x ($(dim(x))) not equal to that of y ($(dim(y)))"
```

Comment:
:+1: Since we are about to change it, maybe we could make it a proper sentence and follow the Julia docs and start with a lower-case letter?
```suggestion
                "dimensionality of x ($(dim(x))) is not equal to that of y ($(dim(y)))"
```